src/main/java/net/neoforged/neoforge/fluids/potion/PotionFluidHandlerItem.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids.potion;
+
+import net.minecraft.core.component.DataComponents;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.ItemLike;
+import net.neoforged.neoforge.common.NeoForgeMod;
+import net.neoforged.neoforge.fluids.FluidStack;
+import net.neoforged.neoforge.fluids.FluidUtil;
+import net.neoforged.neoforge.fluids.capability.IFluidHandler;
+import net.neoforged.neoforge.fluids.capability.IFluidHandlerItem;
+
+// Copy of FluidBucketWrapper to allow differing capacities and empty containers
```

Comment:
Instead of copying, why not either extend it or extract the shared logic into a base class? Duplicate code tends to be a bad idea if it can be avoided.

There was just one method I had to override in my version of this IIRC.